File: python/NeetCode150/10_is_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s) < 2: # String is not big enough to not be a palindrome
            return True
        # Clean the string
        cleaned_s = ""
        for c in s:
            if ord('A') <= ord(c) and ord(c) <= ord('Z'):
                cleaned_s+=c
            elif ord('a') <= ord(c) and ord(c) <= ord('z'):
                cleaned_s+=c
            elif ord('0') <= ord(c) and ord(c) <= ord('9'):
                cleaned_s+=c
        cleaned_s = cleaned_s.lower()
        print(cleaned_s)
        
        if len(cleaned_s) < 2: # String is not big enough to not be a palindrome
            return True

        # We now have a cleaned string
        print((len(cleaned_s) // 2) + 1)
        for i in range(0, (len(cleaned_s) // 2) + 1):
            if cleaned_s[i] != cleaned_s[-i - 1]:
                return False
        return True

File: java/NeetCode150/10_is_palindrome.java
class Solution {
    public boolean isPalindrome(String s) 
    {
        String cleaned_s = "";
        for(int i=0; i<s.length(); i++)
        {
            int c = (int)s.charAt(i);
            if((int)'A' <= c && (int)'Z' >= c)
            {
                cleaned_s = cleaned_s + (char)(c + (int)'a' - (int)'A');
            }
            else if((int)'0' <= c && (int)'9' >= c)
            {
                cleaned_s = cleaned_s + (char)(c);
            }
            else if((int)'a' <= c && (int)'z' >= c)
            {
                cleaned_s = cleaned_s + (char)(c);
            }
        }
        System.out.println(cleaned_s);
        if(cleaned_s.length() < 2)
        {
            return true;
        }
        for(int i = 0; i < (cleaned_s.length() / 2) + 1; i++)
        {
            if(cleaned_s.charAt(i) != cleaned_s.charAt(cleaned_s.length() - i - 1))
            {
                return false;
            }
        }
        return true;
    }
}
